### PR TITLE
simplify reconciler

### PR DESCRIPTION
**Pull request by zw0610**
To support `GenericJob` in tf-operator, a kind of Job independent from any deep learning frameworks, this pull request introduces the following modification to `reconciler.v1` package:

1. Simplify multiple reconciler names (by dropping redundant words like `Kubeflow`)
2. Fix `PodList`, `ServiceList` converting issue
3. Fix default `GetReconcilerName` returned value
4. Simplify `reconciler.v1` package by moving the `KubeflowReconciler` to tf-operator https://github.com/kubeflow/tf-operator/pull/1398
3. Simplify `TestReconciler` construction function

### CODE REVIEW

File: pkg/reconciler.v1/common/service.go
Comment (terrytangyuan) [inline pkg/reconciler.v1/common/service.go:45]: Still mentions KubeflowJob here

File: pkg/reconciler.v1/common/README.md
Comment (terrytangyuan) [inline pkg/reconciler.v1/common/README.md:9]: What about something like Training or TrainingJob since we are always dealing with model trainings here. Job seems very generic.

File: pkg/reconciler.v1/common/service.go
Comment (zw0610) [reply pkg/reconciler.v1/common/service.go]: Thanks for pointing out. I changed `KubeflowJob` to `generic training job`.

File: pkg/reconciler.v1/common/README.md
Comment (zw0610) [reply pkg/reconciler.v1/common/README.md:9]: `TrainingJob` sounds good to me. As the common library is designed for all kinds of controllers, I would suggest renaming the `GenericJob` to `TrainingJob` in the pull request I mentioned above for tf-operator (https://github.com/kubeflow/tf-operator/pull/1398). What do you think?

Comment (terrytangyuan) [reply pkg/reconciler.v1/common/README.md:9]: SGTM

**Comment by Jeffwan** [comment]
Sorry for the delay. Let me merge #163 first before accepting any new changes. This is really annoying.

**Comment by zw0610** [comment]
@Jeffwan could we move this pr forward as the revert string issue is resolved now.

**Review by terrytangyuan — APPROVED** [review]
LGTM
/assign @Jeffwan

File: pkg/reconciler.v1/common/utils.go
Comment (Jeffwan) [inline pkg/reconciler.v1/common/utils.go:28]: "Kubeflow Reconciler" won't work because it's not a word?

File: pkg/reconciler.v1/common/interface.go
Comment (Jeffwan) [inline pkg/reconciler.v1/common/interface.go:244]: What's the motivation to move `KubeflowReconcilerInterface` to `training-operator`?

File: pkg/reconciler.v1/common/service.go
Comment (Jeffwan) [inline pkg/reconciler.v1/common/service.go:89]: em. what's the problem of original way?

File: pkg/reconciler.v1/common/service.go
Comment (Jeffwan) [inline pkg/reconciler.v1/common/service.go:88]: Can we do `var svcs []*corev1.Service`?  it's a little bit weird to explicitly assign nil here.

**Comment by Jeffwan** [comment]
@zw0610 sorry for late response. Please have a check on above comments

File: pkg/reconciler.v1/common/utils.go
Comment (zw0610) [reply pkg/reconciler.v1/common/utils.go:28]: Yes. When generating the default labels, the returned string from `GetReconcilerName` will be used as the value for key `commonv1.OperatorNameLable` (`""training.kubeflow.org/operator-name""`).

File: pkg/reconciler.v1/common/interface.go
Comment (zw0610) [reply pkg/reconciler.v1/common/interface.go:244]: When writing the test code, `test_reconciler`, which could be considered as one example of how the `common.reconciler` will be used, I found that:

1. Such interface will not be used for any abstraction when implementing a real reconciler
2. Embedding modularized interfaces in `KubeflowReconcilerInterface` conflicts embedding them in the real reconciler struct.

File: pkg/reconciler.v1/common/service.go
Comment (zw0610) [reply pkg/reconciler.v1/common/service.go:88]: sure. also changed for `var pods []*corev1.Pod`.

File: pkg/reconciler.v1/common/service.go
Comment (zw0610) [reply pkg/reconciler.v1/common/service.go:89]: It seems the old way will not give a new address for `svc` so when we take the pointer of `svc`, it remains the same. Please see the screenshot below.

Screen Shot 2021-12-12 at 20 14 53

File: pkg/reconciler.v1/common/interface.go
Comment (Jeffwan) [reply pkg/reconciler.v1/common/interface.go:244]: I think it make sense to me. current testing integration between common and training-operator is a headache.

File: pkg/reconciler.v1/common/service.go
Comment (Jeffwan) [reply pkg/reconciler.v1/common/service.go:89]: I see. yeah.  This pointer points to the shared memory location that each of the slice items are being copied into.
All the pointers created using `&svc` point to the same memory location.

Current way looks good to me.

**Comment by Jeffwan** [comment]
/lgtm
/hold

This change looks good to me. Let's see if anyone else have further feedbacks. /cc @kubeflow/wg-training-leads

**Review by Jeffwan — APPROVED** [review]

**Comment by zw0610** [comment]
Any feedback before we can merge this pr? @kubeflow/wg-training-leads

**Pull request merged**
